src/Api/API.php

```diff
@@ -1681,8 +1679,9 @@ protected function searchItems($itemtype, $params = [])
 
        // probably a sql error
         if (!isset($rawdata['data']) || count($rawdata['data']) === 0) {
+            $sql_errors = implode("\n", Profile::getCurrent()->getSQLErrors());
```

Comment:
I think it could be better to store errors in a `protected array $errors = []` property of the `DBmysql` class and get them using a `$DB->getErrors()` method.

It would make SQL errors accessibles without having to activate the debug mode.

Reply:
I thought about it while working on this part but didn't comment on it. However, do we really want to expose the SQL error to the user, even if it doesn't contain the full query? It doesn't seem like something the user has the power to fix.

Reply:
I would personally not have exposed the SQL errors.

Maybe the message could be made generic, like `An unexpected error occurred`.

@orthagh Do you think it is really usefull to output SQL error in the API?

Reply:
We just discussed about it IRL with @orthagh and we agreed that the error details could be removed. A generic message is sufficient.

Reply:
Can't remember why, honnestly it was a long time ago I did this code, and at this time, exposing sql was not considered that critical.

Please remove these things. I don't have anything against the removal